lib/tasks/lumen.rake

```diff
@@ -255,6 +255,29 @@ namespace :lumen do
     end
   end
 
+  desc 'Populate domain_counts table'
+  task populate_domain_count: :environment do
+
+    def give_domains_from_urls(infringing_urls_list)
+      infringing_urls_list.map{ |x| Addressable::URI.parse(x.url).domain }.uniq
+    end
+
+    domain_count_map = {}
+    Notice.find_in_batches do |notices|
+      notices.each do |notice|
+        give_domains_from_urls(notice.infringing_urls).each do |domain|
+          unless domain_count_map.key?(domain)
+            domain_count_map[domain] = 0
+          end
+          domain_count_map[domain] += 1
+        end
+      end
+    end
+    domain_count_map.each do |key, val|
```

Comment:
I just came across this thing that suggests that wrapping this sort of `each...do` in a transaction will make it much faster: https://medium.com/@andresakata/rails-performance-tips-494824a7ded7